pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -367,7 +387,22 @@ public boolean registerNamespace(String namespace, boolean ensureOwned) throws P
                     if (LOG.isDebugEnabled()) {
                         LOG.debug("Namespace bundle {} already owned by {} ", bundle, nsData);
                     }
-                    future.complete(Optional.of(new LookupResult(nsData.get())));
+                    // find the target
+                    if (StringUtils.isNotBlank(advertisedListenerName)) {
+                        AdvertisedListener listener = nsData.get().getAdvertisedListeners().get(advertisedListenerName);
+                        if (listener == null) {
+                            future.completeExceptionally(
+                                    new PulsarServerException("the broker do not have " + advertisedListenerName + " listener"));
+                            return;
+                        } else {
+                            future.complete(Optional.of(new LookupResult(
+                                    nsData.get().getHttpUrl(), nsData.get().getHttpUrlTls(), listener.getBrokerServiceUrl().toString(),
```

Comment:
Why do we get `HttpUrl` and `HttpUrlTls` from NamespaceBundle but get `brokerServiceUrl` and `brokerServiceUrlTls`?

Reply:
currently we only concern pulsar protocol here, the HttpUrl and HttpUrlTls behavior as before